### PR TITLE
added brightness for oled display ssd1306

### PR DESCRIPTION
**Pull request by waiet**
Added brightness controll for oled display ssd_1306.
Tested with i2c ssd1306 1.3" and spi ssd1306 2.42"

### CODE REVIEW

**Comment by davericher** [comment]
Would love to see this go through, could you possibly fix the linting error so it passes the gates?

**Comment by glmnet** [comment]
@waiet thanks for contributing, can you check the build check failure of Travis and fix your code so it is happy for commit?

File: esphome/components/ssd1306_base/ssd1306_base.h [hunk 1]
```diff
@@ -29,6 +29,8 @@ class SSD1306 : public PollingComponent, public display::DisplayBuffer {
   void set_model(SSD1306Model model) { this->model_ = model; }
   void set_reset_pin(GPIOPin *reset_pin) { this->reset_pin_ = reset_pin; }
   void set_external_vcc(bool external_vcc) { this->external_vcc_ = external_vcc; }
+  void set_brightness_(float brightness) { this->brightness_ = brightness; }
+//  void set_brightness_(optional<float> brightness) { this->brightness_ = brightness; }
```
Comment (OttoWinter) [inline esphome/components/ssd1306_base/ssd1306_base.h]: Remove commented out code please.

File: esphome/components/ssd1306_base/ssd1306_base.cpp [hunk 2]
```diff
@@ -100,7 +97,7 @@ void SSD1306::setup() {
     this->command(0xF1);
 
   this->command(SSD1306_COMMAND_SET_VCOM_DETECT);
-  this->command(0x40);
+  this->command(0x00);
```
Comment (OttoWinter) [inline esphome/components/ssd1306_base/ssd1306_base.cpp]: What is this change for? Is it required for different brightness settings? Will this prevent some brightness settings from working?

File: esphome/components/ssd1306_base/ssd1306_base.cpp [hunk 3]
```diff
@@ -79,11 +79,8 @@ void SSD1306::setup() {
     case SH1106_MODEL_128_64:
     case SSD1306_MODEL_64_48:
     case SH1106_MODEL_64_48:
-      if (this->external_vcc_)
-        this->command(0x9F);
-      else
-        this->command(0xCF);
-      break;
+        this->command(int(255 * (this->brightness_)));
```
Comment (OttoWinter) [inline esphome/components/ssd1306_base/ssd1306_base.cpp]: To fix the lint error, dedent this and the next line by 2 spaces.

**Comment by davericher** [comment]
@glment you got this? Would really like to see this implemented, if your not up to the corrections I can take over.

File: esphome/components/ssd1306_base/ssd1306_base.cpp (hunk 2 above)
Comment (glmnet) [reply esphome/components/ssd1306_base/ssd1306_base.cpp]: I have an SSD1306 128x64, with this change the lowest brightness are more lower, in any case 0% is visible. Kind of works.

**Pull request closed without merging**